### PR TITLE
Add track for sub-bucket-aggs

### PR DESCRIPTION
**Pull request by nik9000**
Adds a rally track specifically for testing the performance of the terms
agg as I'll be doing some work on it. In particular this focuses on
numeric terms because the first phase of my work only touches them.

Relates to https://github.com/elastic/elasticsearch/pull/55873

### CODE REVIEW

**Comment by dliappis** [comment]
Adding this link for general awareness re: aggs tuning https://www.elastic.co/blog/improving-the-performance-of-high-cardinality-terms-aggregations-in-elasticsearch

**Comment by nik9000** [comment]
I have an addition to this for https://github.com/elastic/elasticsearch/pull/56921 which makes it more like a track for "sort of complex aggs". I can certainly hold off on pushing it so as to not get in the way of review for this though.

**Comment by nik9000** [comment]
I've added a couple more things to check now that I'm further in this process.

**Comment by hub-cap** [comment]
Ill check them out. I tested the original 3 queries, and found that you can lower the interval on 2 of them and it still works reliably. 

```
+          "operation": "numeric_terms_numeric_terms",
+          "clients": 1,
+          "warmup-iterations": 10,
+          "iterations": 50,
+          "target-interval": 2
+        },
+        {
+          "operation": "date_histo_numeric_terms",
+          "clients": 1,
+          "warmup-iterations": 10,
+          "iterations": 50,
+          "target-interval": 3
```

I will test w/ the new queries before going and recommending that you change those, though.

**Comment by nik9000** [comment]
@hub-cap I've made your changes locally and am playing with some more.

**Comment by hub-cap** [comment]
cool, im going to also tweak the new tasks youve added, to see if we can get a bit more throughput out of them

**Comment by nik9000** [comment]
> cool, im going to also tweak the new tasks youve added, to see if we can get a bit more throughput out of them

I just finished a run with the new throughput you suggested and they choke up Elasticsearch on my desktop a bit:
```
|                                                 Min Throughput |          keyword_terms_numeric_terms |        0.23 |  ops/s |
|                                              Median Throughput |          keyword_terms_numeric_terms |        0.24 |  ops/s |
|                                                 Max Throughput |          keyword_terms_numeric_terms |        0.24 |  ops/s |
|                                        50th percentile latency |          keyword_terms_numeric_terms |     9482.96 |     ms |
|                                        90th percentile latency |          keyword_terms_numeric_terms |     10416.5 |     ms |
|                                       100th percentile latency |          keyword_terms_numeric_terms |     10693.8 |     ms |
|                                   50th percentile service time |          keyword_terms_numeric_terms |     4056.17 |     ms |
|                                   90th percentile service time |          keyword_terms_numeric_terms |     4140.89 |     ms |
|                                  100th percentile service time |          keyword_terms_numeric_terms |     4369.19 |     ms |
|                                                     error rate |          keyword_terms_numeric_terms |           0 |      % |
|                                                 Min Throughput |          numeric_terms_numeric_terms |        0.45 |  ops/s |
|                                              Median Throughput |          numeric_terms_numeric_terms |        0.45 |  ops/s |
|                                                 Max Throughput |          numeric_terms_numeric_terms |        0.46 |  ops/s |
|                                        50th percentile latency |          numeric_terms_numeric_terms |     9209.06 |     ms |
|                                        90th percentile latency |          numeric_terms_numeric_terms |     12699.1 |     ms |
|                                       100th percentile latency |          numeric_terms_numeric_terms |     13593.9 |     ms |
|                                   50th percentile service time |          numeric_terms_numeric_terms |     2178.83 |     ms |
|                                   90th percentile service time |          numeric_terms_numeric_terms |     2195.56 |     ms |
|                                  100th percentile service time |          numeric_terms_numeric_terms |     2294.51 |     ms |
|                                                     error rate |          numeric_terms_numeric_terms |           0 |      % |
```

It doesn't fall over but it ain't so happy.

**Comment by hub-cap** [comment]
oh, indeed they do choke up on the local, im using a multiple machine setup. Ill circle back to find out which of these scenarios is what we should commit to the repo.

**Comment by nik9000** [comment]
> Looks great, but names need to be kebap case and not snake case

I've stabbed my words.

File: noaa/operations/default.json [hunk 1]
```diff
@@ -713,5 +713,313 @@
...
+                  "buckets": 20,
+                  "time_zone": "America/New_York"
+                }
+              }
+            }
+          }
+        }
+      }
     }
-
```
Comment (hub-cap) [inline noaa/operations/default.json:717]: trailing newline pls

Comment (hub-cap) [reply noaa/operations/default.json:717]: disregard i dumb.

Comment (nik9000) [reply noaa/operations/default.json:717]: Me too! What a coincidence!

**Pull request merged**